cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -127,34 +142,75 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
+    the shortest sequence and the first segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[:limit], str_2=sequence_2[:limit])
+
+
+def get_hamming_distance_index_2(sequence_1: str, sequence_2: str) -> int:
+    """Get the hamming distance between two index 2 sequences.
+    In the case that one sequence is longer than the other, the distance is calculated between
+    the shortest sequence and the last segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[-limit:], str_2=sequence_2[-limit:])
+
+
+def adapt_barcode_mismatch_values_for_sample(
+    sample_to_update: FlowCellSampleNovaSeqX, samples: List[FlowCellSampleNovaSeqX]
 ) -> None:
-    """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
-    number  of index reads, and takes the reverse complement of index 2 in case of the new
-    novaseq software control version (1.7) in combination with the new reagent kit
-    (version 1.5).
-    """
-    LOG.info("Fix so that all indexes are in the correct format")
-    reverse_complement: bool = is_reverse_complement(run_parameters=run_parameters)
+    """Updates the barcode mismatches for both indexes of a FlowCellSampleNovaSeqX."""
+    index_1_sample_to_update, index_2_sample_to_update = get_index_pair(sample=sample_to_update)
     for sample in samples:
-        index1, index2 = sample.index.split("-")
-        index1: str = index1.strip()
-        index2: str = index2.strip()
-        index_length = len(index1)
-        if is_padding_needed(
-            index_cycles=run_parameters.index_length, sample_index_length=index_length
+        if sample_to_update.sample_id == sample.sample_id:
+            continue
+        index_1, index_2 = get_index_pair(sample=sample)
+        if (
+            get_hamming_distance_index_1(sequence_1=index_1_sample_to_update, sequence_2=index_1)
+            < MINIMUM_HAMMING_DISTANCE
+        ):
+            sample_to_update.barcode_mismatches_1 = 0
+        if (
+            get_hamming_distance_index_2(sequence_1=index_2_sample_to_update, sequence_2=index_2)
+            < MINIMUM_HAMMING_DISTANCE
         ):
-            LOG.debug("Padding indexes")
-            index1 = pad_index_one(index_string=index1)
-            index2 = pad_index_two(index_string=index2, reverse_complement=reverse_complement)
-        if reverse_complement:
-            index2 = get_reverse_complement_dna_seq(index2)
-        sample.index = index1
-        sample.index2 = index2
+            sample_to_update.barcode_mismatches_2 = 0
 
 
-def is_dual_index(index: str) -> bool:
-    """Determines if an index in the raw sample sheet is dual index or not."""
-    return "-" in index
+def adapt_indexes_for_sample(
```

Comment:
Is there a more descriptive name than `adapt`?

Reply:
Updated docstring